File: mobile/widgets/AuraWidget.swift
import WidgetKit
import SwiftUI

// Aura iOS WidgetKit Extension
// Enables 1-Tap Voice Purge and 30s Panic Reset directly from iOS Home Screen & Lock Screen

struct AuraWidgetEntry: TimelineEntry {
    let date: Date
    let clarityScore: Int
    let nocturnalWorriesResolvedPct: Int
    let pendingMorningDeliberations: Int
}

struct AuraWidgetProvider: TimelineProvider {
    func placeholder(in context: Context) -> AuraWidgetEntry {
        AuraWidgetEntry(date: Date(), clarityScore: 94, nocturnalWorriesResolvedPct: 82, pendingMorningDeliberations: 1)
    }

    func getSnapshot(in context: Context, completion: @escaping (AuraWidgetEntry) -> Void) {
        let entry = AuraWidgetEntry(date: Date(), clarityScore: 94, nocturnalWorriesResolvedPct: 82, pendingMorningDeliberations: 1)
        completion(entry)
    }

    func getTimeline(in context: Context, completion: @escaping (Timeline<AuraWidgetEntry>) -> Void) {
        let currentDate = Date()
        let refreshDate = Calendar.current.date(byAdding: .minute, value: 30, to: currentDate)!
        let entry = AuraWidgetEntry(date: currentDate, clarityScore: 94, nocturnalWorriesResolvedPct: 82, pendingMorningDeliberations: 0)
        let timeline = Timeline(entries: [entry], policy: .after(refreshDate))
        completion(timeline)
    }
}

struct AuraWidgetEntryView: View {
    var entry: AuraWidgetProvider.Entry

    var body: some View {
        ZStack {
            Color(red: 0.04, green: 0.04, blue: 0.06)
                .edgesIgnoringSafeArea(.all)

            VStack(alignment: .leading, spacing: 6) {
                HStack {
                    Text("AURA")
                        .font(.system(size: 11, weight: .black, design: .monospaced))
                        .foregroundColor(.cyan)
                    Spacer()
                    Circle()
                        .fill(Color.cyan)
                        .frame(width: 6, height: 6)
                        .shadow(color: .cyan, radius: 4)
                }

                Spacer()

                Text("\(entry.nocturnalWorriesResolvedPct)%")
                    .font(.system(size: 26, weight: .bold, design: .rounded))
                    .foregroundColor(.white)

                Text("Dissolved by morning")
                    .font(.system(size: 9, weight: .medium))
                    .foregroundColor(.gray)

                HStack(spacing: 6) {
                    // Deep link to Voice Purge
                    Link(destination: URL(string: "aura://voice-purge")!) {
                        HStack {
                            Image(systemName: "mic.fill")
                                .font(.system(size: 9))
                            Text("Purge")
                                .font(.system(size: 9, weight: .bold))
                        }
                        .padding(.vertical, 4)
                        .padding(.horizontal, 8)
                        .background(Color.cyan.opacity(0.2))
                        .foregroundColor(.cyan)
                        .cornerRadius(12)
                    }

                    // Deep link to SOS Panic Reset
                    Link(destination: URL(string: "aura://panic-reset")!) {
                        HStack {
                            Image(systemName: "wind")
                                .font(.system(size: 9))
                            Text("SOS")
                                .font(.system(size: 9, weight: .bold))
                        }
                        .padding(.vertical, 4)
                        .padding(.horizontal, 8)
                        .background(Color.red.opacity(0.2))
                        .foregroundColor(.red)
                        .cornerRadius(12)
                    }
                }
            }
            .padding(12)
        }
    }
}

@main
struct AuraWidget: Widget {
    let kind: String = "AuraWidget"

    var body: some WidgetConfiguration {
        StaticConfiguration(kind: kind, provider: AuraWidgetProvider()) { entry in
            AuraWidgetEntryView(entry: entry)
        }
        .configurationDisplayName("Aura Quick Sanctuary")
        .description("1-Tap Voice Purge into Vapor and 30-Second Panic Sensory Reset.")
        .supportedFamilies([.systemSmall, .accessoryCircular, .accessoryRectangular])
    }
}

File: mobile/widgets/AuraWidget.kt
package com.auraclarity.widget

import android.content.Context
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.glance.*
import androidx.glance.action.actionStartActivity
import androidx.glance.action.clickable
import androidx.glance.appwidget.GlanceAppWidget
import androidx.glance.appwidget.GlanceAppWidgetReceiver
import androidx.glance.appwidget.provideContent
import androidx.glance.layout.*
import androidx.glance.text.FontWeight
import androidx.glance.text.Text
import androidx.glance.text.TextStyle

/**
 * Aura Android Glance Home Screen & Lock Screen Widget.
 * Offers 1-Tap Zero-Effort Voice Purge and 30s SOS Sensory Panic Reset.
 */
class AuraGlanceWidget : GlanceAppWidget() {

    override suspend fun provideGlance(context: Context, id: GlanceId) {
        provideContent {
            GlanceTheme {
                AuraWidgetContent()
            }
        }
    }

    @Composable
    private fun AuraWidgetContent() {
        Column(
            modifier = GlanceModifier
                .fillMaxSize()
                .background(Color(0xFF090A0F))
                .padding(12.dp),
            verticalAlignment = Alignment.Vertical.CenterVertically,
            horizontalAlignment = Alignment.Horizontal.CenterHorizontally
        ) {
            Text(
                text = "AURA",
                style = TextStyle(
                    color = Color(0xFF4CD7F6),
                    fontSize = 12.sp,
                    fontWeight = FontWeight.Bold
                )
            )

            Spacer(modifier = GlanceModifier.height(4.dp))

            Text(
                text = "82%",
                style = TextStyle(
                    color = Color.White,
                    fontSize = 24.sp,
                    fontWeight = FontWeight.Bold
                )
            )

            Text(
                text = "Worries dissolved by morning",
                style = TextStyle(
                    color = Color(0xFF94A3B8),
                    fontSize = 9.sp
                )
            )

            Spacer(modifier = GlanceModifier.height(8.dp))

            Row(
                modifier = GlanceModifier.fillMaxWidth(),
                horizontalAlignment = Alignment.Horizontal.CenterHorizontally
            ) {
                // 1-Tap Voice Purge Trigger
                Box(
                    modifier = GlanceModifier
                        .background(Color(0xFF1E293B))
                        .padding(horizontal = 8.dp, vertical = 4.dp)
                ) {
                    Text(
                        text = "🎤 Voice Purge",
                        style = TextStyle(color = Color(0xFF38BDF8), fontSize = 10.sp, fontWeight = FontWeight.Bold)
                    )
                }

                Spacer(modifier = GlanceModifier.width(6.dp))

                // SOS 30s Panic Reset Trigger
                Box(
                    modifier = GlanceModifier
                        .background(Color(0xFF3B1D28))
                        .padding(horizontal = 8.dp, vertical = 4.dp)
                ) {
                    Text(
                        text = "⚡ 30s SOS",
                        style = TextStyle(color = Color(0xFFFDA4AF), fontSize = 10.sp, fontWeight = FontWeight.Bold)
                    )
                }
            }
        }
    }
}

class AuraGlanceReceiver : GlanceAppWidgetReceiver() {
    override val glanceAppWidget: GlanceAppWidget = AuraGlanceWidget()
}
